### PR TITLE
Fix doc/Makefile that can't build man because of doc/man folder exists

### PR DESCRIPTION
**Pull request by shibukawa**

### CODE REVIEW

**Comment by tk0miya** [comment]
Hmm... this is also reproduced at new sphinx project.

```
$ sphinx-quickstart test
$ cd test
$ touch html
$ make html
make: `html' is up to date.
```

I believe this is a bug of new "make mode".

@shimizukawa We have to fix this until next release. Or I will change the default make mode to old one.

**Comment by tk0miya** [comment]
The manual of GNU make says:

> The implicit rule search (see Implicit Rules) is skipped for .PHONY targets. This is why declaring a target as .PHONY is good for performance, even if you are not worried about the actual file existing.

https://www.gnu.org/software/make/manual/make.html#Phony-Targets

Because of that, anything pattern (`%`) could not work for phony targets. So `make` command could not invoke sphinx if there are files or directories having same name with builders (like `html`, `man` and so on).

**Comment by shimizukawa** [comment]
@birkenfeld do you have any opinion? or any idea to fix the problem of make-mode?

**Comment by tk0miya** [comment]
I posted #3029 for the default mode of quickstart.

About this issue, both merging or rejecting is ok for me.
I think sphinx/doc is not a manual. So nobody really needs this change.

**Comment by shimizukawa** [comment]
+1 for merging

**Comment by tk0miya** [comment]
Okay. @shibukawa could you rebase this please? Then I will merge right away.

**Comment by shibukawa** [comment]
rebased

File: CHANGES [hunk 1]
```diff
@@ -38,6 +38,7 @@ Bugs fixed
   target using its caption automatially.
 * #2962: latex: missing label of longtable
 * #2968: autodoc: show-inheritance option breaks docstrings
+* #2936: Fix doc/Makefile that can't build man because doc/man exists
```
Comment (tk0miya) [inline CHANGES]: Thank you for rebasing. But this entry should move to 1.4.9

**Review by tk0miya — CHANGES_REQUESTED** [review]

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Merged. Thank you always!

**Comment by jfbu** [comment]
I am sure I do not understand exactly what is at stake but can't you just include one giant single

```
.PHONY: help man clean html dirhtml epub <etc>...
```

into  the `Makefile.new_t` template with all builders known to Sphinx ?

**Comment by jfbu** [comment]
sorry I get it now. silly question.

**Comment by jfbu** [comment]
would something like this make senses

```
TARGETSWITHBUILDERNAMES = help clean man html latex

.PHONY: $(TARGETSWITHBUILDERNAMES)

$(TARGETSWITHBUILDERNAMES):
    @$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)

# Catch-all target: route all unknown targets to Sphinx using the new
# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).

%:
    @$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

**Comment by tk0miya** [comment]
The make-mode was introduced not to maintainance Makefile even if we added new builder to Sphinx.
But your example has a list like `TARGETSWITHBUILDERNAMES`.

I know it is much simple thank current one. but it could not replace make-mode.

**Pull request merged**